description/src/transformer.rs

```diff
@@ -70,3 +72,49 @@ where
         Ok(r)
     }
 }
+
+/// Returns true for types where recursion should continue, instead of being stopped when recursion in being detected.
+///
+/// ## Background:
+///
+/// There is a problem in generating recursive type descriptions:
+/// Suppose we have the following setup:
+/// ```rust
+/// struct A {
+///     bees: Vec<B>
+/// }
+///
+/// struct B {
+///     id: u8,
+///     others: Vec<B>
+/// }
+/// ```
+/// This could be described as:
+/// ```txt,no_run
+/// struct A {
+///     bees: Vec<struct B {
+///         id: u8,
+///         others: Vec<B>
+///     }>
+/// }
+/// ```
+/// But the recursive resolving would get stuck in the middle, reporting recursion.
+/// This is because Vec<B> needs to be mapped to different strings, so the simple cache lookup is not viable.
+/// The solution to this is, to just let some container types like Vec do recursion while others can't.
+///
+/// # Warning
+///
+/// The safety of the following logic relies on the assumption that ultimately everything resolves down to a primitive or a struct/enum that is in the cache.
+/// It basically just returns true for generic wrapper types.
```

Comment:
I'm struggling to really understand this `recursion_should_continue` thing; perhaps because I don't grasp the problem well enough!

My understanding so far from reading the above is:

We have type info for types like

```rust
struct A { bees: Vec<B> } 
struct B { id: u8, others: Vec<B> }
```

We'd like to print a description like:

```rust
struct A { 
    bees: Vec<struct B {
        id: u8,
        others: Vec<B>
    }
> 
} 
```

So the first time `Vec<B>` is encountered, we expand B. The second time `Vec<B>` is encountered, we'll be recursing here, and want to instead print `Vec<B>` rather than expanding it.

Naively, to me it feels like this could be achieved by running some function on recursion that takes in the type info, here the info about B, and wants back some string to use. Since we know we are recursing in order to run this function, the function looks at the type name, B, and returns that string here.

> That is because, if the policy you described was applied here, the type which is seen twice in recursion is Vec<B>

You said this above in response to I think alex suggesting the same. But wouldn't the type that's seen twice in recursion be `B` and not `Vec<B>`? (ie, we would recurse into a Vec again, and then see that the type in the vec was B, and then say oohh wait, I've already encountered B before. We wouldn't go into a vec and instantly decide that the vec itself was resurding would we?)

Reply:
Thanks for explaining this!

The bit I missed was that `Vec<B>` is itself a distinct type, and when we hit the second one we'd trigger the "aah we are recursing" thing, and so need to explicitly opt-in to recursing into this sort of type to print it properly without some special case to print `Vec<B>` or whatever.